modules/database/audit-autoloaded-options/update-autoload.js

```diff
@@ -0,0 +1,45 @@
+/* eslint camelcase: "off" */
+/* global jQuery:false, perflabAutoloadSettings:false */
+
+( function ( $, document ) {
+	$( document ).on( 'click', '.update-autoload', function ( e ) {
+		e.preventDefault();
+		const button = $( this );
+		const optionName = button.data( 'option-name' );
+		const autoload = button.data( 'autoload' );
+		const value = button.data( 'value' );
+		const data = {
+			option_name: optionName,
+			autoload,
+			value,
+		};
+
+		$.ajax( {
+			url:
+				perflabAutoloadSettings.root + 'perflab-aao/v1/update-autoload',
+			method: 'POST',
+			data,
+			success( response ) {
+				if ( response.success ) {
+					button.attr( 'disabled', true );
+				} else {
+					button.append(
+						'<span class=""> ' + response.message + '</span>'
+					);
+					$(
+						'<div style="margin-top:5px;border:1px solid #d63638;padding-left:5px;">' +
+							response.message +
+							'</div>'
+					).insertAfter( button );
+				}
```

Comment:
As far as I can tell, the success callback doesn't handle moving the option to the other table, which creates a weird user experience which would require the user to reload the page to see what it should _actually_ look like after performing the action.

I see two options here:
* Either we fix this by expanding the JS logic. Doing it with AJAX can create a better user experience - but only if done right. The end result should be the same as if you refreshed the page afterwards.
* Or we ditch the AJAX logic and REST endpoint and go with a much faster and easier to implement "action URL" that handles the logic (e.g. via `admin_action_{$action}` hook). This would be _a lot_ easier to implement in a way that doesn't feel weird.

Again, in theory AJAX can make for a smoother experience than a page reload - but not if the AJAX experience is actually sub par in what it does. Worth noting that handling this via AJAX and JS requires some JS code that duplicates functionality which PHP already handles, so there's always a maintenance overhead. Not sure whether it's worth it.

Reply:
In https://github.com/WordPress/performance/pull/1048/commits/781871b617d14ff061c572ceef3a9d48bec3ebab i have update approach.